src/Utilities/StdHelpers.hpp

```diff
@@ -283,3 +283,61 @@ std::string formatted_string(const std::string& fmt, Args... args) {
   }
   return rtn;
 }
+
+// Arithmetic operators for std::array<T, Dim>
+
+template <size_t Dim, typename T>
+inline void operator+=(std::array<T, Dim>& lhs, const std::array<T, Dim>& rhs) {
```

Comment:
Should compute `noexcept`ness

Reply:
Could you elaborate? Do you mean to mark this `noexcept`?

Reply:
No, sorry, I meant that we should check `noexcept(noexcept(lhs[0] += rhs[0]))`, which means we may need to special case `std::array<T, 0>`. I'm hopefully this would count as overloading the function rather than partial function template specialization, otherwise `template <typename T, bool = true>` or, maybe better, `template <typename T, std::nullptr_t = nullptr>` would work to create a second function template, and then have a `Requires<Dim != 0>` in the original. The `Dim == 0` can be marked as `noexcept` Does that explain make sense?